### ISSUE TITLE
Show progress during `check-setup`

### ISSUE THREAD

**Issue by corneliusroemer** [nextstrain/cli#359]
`nextstrain check-setup` takes almost a minute despite having a fast M1 Pro. There's no progress indicator which might be confusing to users who are running it for the first time.

It might be good to give an indication how long this check takes. E.g. "Testing your setup... (this takes about a minute)"

Even better would be some sort of progress indication after each runtime is checked, maybe outputting it or indicating that it has been checked successfully.

<img width="1963" alt="image" src="https://github.com/nextstrain/cli/assets/25161793/6ae2a093-5cad-46df-b9b8-3b9917895b42">

**Comment by tsibley** [comment]
Yeah, `check-setup` hasn't gotten a lot of attention. It could use some!

**Comment by victorlin** [comment]
I didn't realize there was an existing issue for this. I hacked together d4e5a78093c8e0dd22f551902d7b2327b0839fbc a while back and this is what it looks like, before and after:

https://github.com/user-attachments/assets/69095e02-0663-405c-a6f4-e9f87e4a881e

If others think this is a good direction, I can turn it into a proper PR.

**Comment by tsibley** [comment]
@victorlin Generally in favor of that direction. Note that I've made some changes to the setup/runner tests in #407 (896c2a6), but they're straightforward renames and I can rebase on top of your changes for this if needed. Probably I should pull out several of the ancillary changes from that branch into a new one and get them merged before the workflows-as-programs features.
